core/chains/chains.go

```diff
@@ -0,0 +1,137 @@
+package chains
+
+import (
+	"math/big"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+// To add a new chain to this file:
+// 1. Add the global var in the vars list
+// 2. Add the chain ID in the map in the init() function
+// 3. Add a config set in configs.go
+
+// Chain represents a blockchain with a unique Chain ID
+type Chain struct {
+	id      *big.Int
+	config  ChainSpecificConfig
+	logOnce sync.Once
+}
+
+func (c *Chain) setChainID(id int64) {
+	c.id = big.NewInt(id)
+}
+
+func (c *Chain) ID() *big.Int {
+	return c.id
+}
+
+func (c *Chain) Config() ChainSpecificConfig {
+	if !c.config.set {
+		c.logOnce.Do(func() {
+			logger.Warnf("chain with ID %s does not have a chain-specific config, using fallback config instead", c.ID())
+		})
+		return FallbackConfig
+	}
+	return c.config
+}
+
+// IsArbitrum returns true if the chain is arbitrum mainnet or testnet
+func (c *Chain) IsArbitrum() bool {
+	if c == nil {
```

Comment:
hm why some methods check `c == nil` while others do not? could be consistent, or maybe explained somewhere

Reply:
Good question. I'll make it consistent.